prow/pod-utils/decorate/podspec.go

```diff
@@ -0,0 +1,163 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package decorate
+
+import (
+	"encoding/json"
+	"fmt"
+	"sort"
+	"strconv"
+
+	"k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"k8s.io/test-infra/prow/kube"
+)
+
+const (
+	JobNameEnv   = "JOB_NAME"
+	JobSpecEnv   = "JOB_SPEC"
+	JobTypeEnv   = "JOB_TYPE"
+	ProwJobIdEnv = "PROW_JOB_ID"
+
+	BuildIdEnv = "BUILD_ID"
+	// Deprecated, will be removed in the future.
+	ProwBuildIdEnv = "BUILD_NUMBER"
+	// Deprecated, will be removed in the future.
+	JenkinsBuildIdEnv = "buildId"
+
+	RepoOwnerEnv   = "REPO_OWNER"
+	RepoNameEnv    = "REPO_NAME"
+	PullBaseRefEnv = "PULL_BASE_REF"
+	PullBaseShaEnv = "PULL_BASE_SHA"
+	PullRefsEnv    = "PULL_REFS"
+	PullNumberEnv  = "PULL_NUMBER"
+	PullPullShaEnv = "PULL_PULL_SHA"
+)
+
+// EnvForSpec returns a mapping of environment variables
+// to their values that should be available for a job spec
+func EnvForSpec(spec JobSpec) (map[string]string, error) {
+	env := map[string]string{
+		JobNameEnv:   spec.Job,
+		BuildIdEnv:   spec.BuildId,
+		ProwJobIdEnv: spec.ProwJobId,
+		JobTypeEnv:   string(spec.Type),
+	}
+
+	// for backwards compatibility, we provide the build ID
```

Comment:
Can we put a plan in place to retire both of these?

Ideally we only have one thing, and ideally we don't have agent specific code in here.